schema/common.json

```diff
@@ -4,6 +4,103 @@
   "description": "Serverless Workflow specification - common schema",
   "type": "object",
   "definitions": {
+    "expressionMarkup": {
+      "type": "string",
+      "pattern": "^\\${.*}$"
+    },
+    "jsonpointer": {
+      "$ref": "#/definitions/expressionMarkup"
+    },
+    "functionMarkup": {
+      "type": "string",
+      "pattern": "^\\$fn{.*}$"
```

Comment:
This is confusing to me honestly. We have an expression definition which is assumed its contents is an expression then we have a function markup which assumes its content is reference to expression function. 

the diff from what i can tell is 
"${ expresion}"
"$fn{ string }"

i think just doing the way we have now:
"${ expression }" 
"${ fn:string }"

is same thing really no reason to change and it makes more sense to me at least.

Reply:
i do like the use of pattern tho.we can add patterns to existing stuff to make it more clear when writing DSL in  IDEs

Reply:
when we markup expressions with `${ }` and allow `${ fn:Function }`, then we first say that everything inside `${}` should be interpreted by the expression language, which may be JQ or something else, meaning you don't know how to interpret what's inside `${}` and it's up to the interpreter to compute that. But then we say that if fn:Function appears inside the markup, we don't care what the interpreter would do with it but instead, we invoke a named expression function.